libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2535,6 +2535,9 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 	if (auto tt = dynamic_cast<TypeType const*>(exprType))
 		if (tt->actualType()->category() == Type::Category::Enum)
 			annotation.isPure = true;
+	if (auto functionType = dynamic_cast<FunctionType const*>(exprType))
+		if (functionType->kind() == FunctionType::Kind::Declaration && memberName == "selector")
```

Comment:
Actually... which member accesses to ``.selector`` are **not** pure? Can we extend this to more of them?

Reply:
If the expression itself is pure, then all of them should.